packages-exp/auth-exp/src/core/user/link_unlink.ts

```diff
@@ -45,3 +52,25 @@ export async function unlink(
   await user.auth._persistUserIfCurrent(user);
   return user;
 }
+
+/**
+ * Internal-only link helper
+ */
+export async function _link(user: User, linkAction: Promise<IdTokenResponse>): Promise<UserCredentialImpl> {
+  return UserCredentialImpl._forOperation(user, externs.OperationType.LINK, await linkAction);
+}
+
+export async function _assertLinkedStatus(
```

Comment:
does this need to be exported? can we have it as a private method only available from _link()?

Reply:
Yeah, it needs to be exported. It's used in `linkWithPhoneNumber()` to check the link status before doing the SMS flow (whereas `_link()` is called _after_ the SMS flow)